cachi2/core/package_managers/yarn/main.py

```diff
@@ -40,14 +40,16 @@ def _resolve_yarn_project(project: Project, output_dir: RootedPath) -> list[Comp
 
     _configure_yarn_version(project)
 
+    if project.is_zero_installs:
```

Comment:
Small difference, but we can probably do this check before _configure_yarn_version. No need to install yarn if we're going to reject the package.

Reply:
I can definitely move the check, although we'd have to move it again in context of #377 which introduces loading Yarn RC defaults using `yarn` itself. There's a non-zero chance I might actually forget to move it as part of #377 especially if more comments pop up during review so if you're not strongly against I'd like to keep this check as is.